spyder/plugins/completion/kite/plugin.py

```diff
@@ -34,16 +34,32 @@ def __init__(self, parent):
         enable_code_snippets = CONF.get('lsp-server', 'code_snippets')
         self.client = KiteClient(None, enable_code_snippets)
         self.kite_process = None
+        self.main_window_visible = False
+        self.kite_initialized = False
+        self.onboarding_shown = False
         self.client.sig_client_started.connect(self.http_client_ready)
         self.client.sig_response_ready.connect(
             functools.partial(self.sig_response_ready.emit,
                               self.COMPLETION_CLIENT_NAME))
+        self.main.sig_setup_finished.connect(self.mainwindow_setup_finished)
 
     @Slot(list)
     def http_client_ready(self, languages):
         logger.debug('Kite client is available for {0}'.format(languages))
         self.available_languages = languages
         self.sig_plugin_ready.emit(self.COMPLETION_CLIENT_NAME)
+        self.kite_initialized = True
+        self._kite_onboarding()
```

Comment:
Is not enough with the signal from the `MainWindow` or maybe sometimes the client isn't ready even when the MainWindow already ended up its setup?